### PR TITLE
Validation for SUPPLEMENTARY_EVIDENCE_WITH_OCR journey classification

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
[Orchestrator to update cases with Supplementary Evidence](https://tools.hmcts.net/jira/browse/BPS-833)

### Change description ###
Added validation for SUPPLEMENTARY_EVIDENCE_WITH_OCR journey classification in callback validator.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java [hunk 1]
```diff
@@ -149,6 +150,15 @@ private CallbackValidations() {
                     switch (classification) {
                         case CLASSIFICATION_SUPPLEMENTARY_EVIDENCE:
                             return Validation.<String, Void>valid(null);
+                        case CLASSIFICATION_SUPPLEMENTARY_EVIDENCE_WITH_OCR:
+                            return hasOcr(theCase)
+                                ? Validation.<String, Void>valid(null)
+                                : Validation.<String, Void>invalid(
+                                format(
+                                    "The 'attach to case' event is not supported for supplementary evidence with OCR "
+                                        + "but not containing OCR data"
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java]: worth consulting with Ade

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java [hunk 2]
```diff
@@ -551,6 +552,50 @@ public void should_succeed_when_classification_is_exception_and_exception_record
...
+    @Test
+    public void should_fail_when_classification_is_supplementary_evidence_with_ocr_does_not_include_ocr() {
+        CallbackRequest callbackRequest =
+            callbackRequestWith(
+                EVENT_ID_ATTACH_TO_CASE,
+                SUPPLEMENTARY_EVIDENCE_WITH_OCR.name(),
+                false
+            );
+
+        ValidatableResponse response =
```
Comment (lgonczar) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java]: Codacy Issue found: [Found 'DU'-anomaly for variable 'response' (lines '585'-'597').](https://app.codacy.com/app/HMCTS/bulk-scan-orchestrator/pullRequest?prid=4550029)

**Comment by lgonczar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-scan-orchestrator/bulk-scan-orchestrator/pullRequest?prid=4550029&bid=15306465)

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java (hunk 2 above)
Comment (lgonczar) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java]: Codacy Issue found: [Avoid unused local variables such as 'response'.](https://app.codacy.com/app/HMCTS/bulk-scan-orchestrator/pullRequest?prid=4550029)

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidationsTest.java [hunk 3]
```diff
@@ -197,8 +199,10 @@ void serviceNameInCaseTypeIdTest(
         return new Object[][]{
             {"Invalid journey classification", createCaseWith(b -> b.data(ImmutableMap.of(JOURNEY_CLASSIFICATION, "invalid_classification"))), false, "Invalid journey classification invalid_classification"},
             {"Valid journey classification(supplementary evidence)", createCaseWith(b -> b.data(ImmutableMap.of(JOURNEY_CLASSIFICATION, "SUPPLEMENTARY_EVIDENCE"))), true, null},
+            {"Valid journey classification(supplementary evidence with ocr)", createCaseWith(b -> b.data(ImmutableMap.of(JOURNEY_CLASSIFICATION, "SUPPLEMENTARY_EVIDENCE_WITH_OCR", OCR_DATA, asList(ImmutableMap.of("f1", "v1"))))), true, null},
+            {"Valid journey classification(supplementary evidence with ocr ocr empty)", createCaseWith(b -> b.data(ImmutableMap.of(JOURNEY_CLASSIFICATION, "SUPPLEMENTARY_EVIDENCE_WITH_OCR", OCR_DATA, emptyList()))), false, "The 'attach to case' event is not supported for supplementary evidence with OCR but not containing OCR data"},
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidationsTest.java]: is there a constant for `"SUPPLEMENTARY_EVIDENCE_WITH_OCR "`?
Similar to `CLASSIFICATION_EXCEPTION` used below.

**Comment by lgonczar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-scan-orchestrator/bulk-scan-orchestrator/pullRequest?prid=4550029&bid=15306465)

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java (hunk 2 above)
Comment (lgonczar) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java]: Codacy Issue found: [Found 'DU'-anomaly for variable 'response' (lines '585'-'597').](https://app.codacy.com/app/HMCTS/bulk-scan-orchestrator/pullRequest?prid=4550029)

Comment (lgonczar) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java]: Codacy Issue found: [Avoid unused local variables such as 'response'.](https://app.codacy.com/app/HMCTS/bulk-scan-orchestrator/pullRequest?prid=4550029)

**Pull request merged**